### PR TITLE
Add support for Symfony 3

### PR DESCRIPTION
**Pull request by SpacePossum**
Hi,

Thanks for all the work on this project!

I would like to use your package with the Symfony 3.x event-dispatcher.
The other changed are:
- replace deprecated PHPUnit methods
- some SCA stuff
- hint in the readme

### CODE REVIEW

**Comment by jmikola** [comment]
@SpacePossum: I suppose the [bundle](https://github.com/jmikola/JmikolaWildcardEventDispatcherBundle) will need to be updated as well. Are you using both packages, or just this base library?

**Comment by SpacePossum** [comment]
I'm using the library only with Silex atm.

**Comment by SpacePossum** [comment]
All green including PHP7 :)
I've updated the test matrix to test with both ^2.3 and 3.x versions of Symfony.
Let me know what you think @jmikola

**Comment by jmikola** [comment]
@SpacePossum: While reviewing this, I encountered some additional edge cases. Most of your changes are incorporated into https://github.com/jmikola/WildcardEventDispatcher/pull/7. I'd appreciate it if you could test that out and let me know if it works all right for you. Thanks!

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "minimum-stability": "dev",
     "require": {
         "php": ">=5.3.2",
-        "symfony/event-dispatcher": "^2.0 || ^3.0"
+        "symfony/event-dispatcher": "^2.3 || ^3.0"
```
Comment (TomasVotruba) [inline composer.json]: Why not bumping to Symfony 2.7 as the lowest LTS?

Comment (jmikola) [inline composer.json]: There likely wasn't any reason to bump to 2.3, as the existing code should still have worked with 2.0. I just happened to miss this when merging the PR and tagging 1.1.0.

If the library doesn't actually depend on newer functionality and BC happens to be preserved, I'm in favor of keeping existing requirements as-is.

Comment (TomasVotruba) [inline composer.json]: I understand there is no technical reason. Yet this slows down continuous upgrade process in PHP ecosystem, since it propagates using unsupported version.

Anyone new coming here might see this as 2.3 is fine to use. Which is not. It doesn't even have security fixes anymore.

Also it suggest it's ok to ask same lower version requirement in other packages.

Same goes for PHP version with no security fixes.

Comment (SpacePossum) [inline composer.json]: I did testing with the code + changes to see what versions of SF would (still) work and found 2.3 the minimal. Bumping the version even higher can be done (of course) but I considered it outside of the scope of the PR.

Comment (TomasVotruba) [inline composer.json]: I can relate to this reason much easily. Thank for explaining @SpacePossum 

@jmikola Would you be open to bump versions with active security support for reasons I've talked about?

Comment (jmikola) [inline composer.json]: @SpacePossum: FWIW, the master branch's test suite still works against Symfony 2.0.4. I changed the lower requirement to "^2.0" and ran the following:

```
$ composer update --prefer-lowest
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing symfony/event-dispatcher (2.2.x-dev)
  - Installing symfony/event-dispatcher (2.0.4)
    Downloading: 100%         

Writing lock file
Generating autoload files

$ phpunit 
PHPUnit 4.8.26 by Sebastian Bergmann and contributors.

.......................S..

Time: 127 ms, Memory: 4.00MB

OK, but incomplete, skipped, or risky tests!
Tests: 26, Assertions: 105, Skipped: 1.
```

> Would you be open to PR with bumping versions of active security support for reasons I've talked about?

@TomasVotruba: I'm not open to modifying this library's dependencies for that purpose. Folks can use a project such as [roave/security-advisories](https://github.com/Roave/SecurityAdvisories), which is regularly updated, to catch insecure versions of packages.

Comment (SpacePossum) [inline composer.json]: I did the check once, maybe after I was done with the PR it turns out it does run on even earlier versions of SF2.
Please note though that 2.0.x is [EOL since 09/2013](http://symfony.com/doc/current/contributing/community/releases.html#version-history). If someone still uses that, that user shouldn't expect much support for it IMHO (they sure won't get it from SF).

Comment (TomasVotruba) [inline composer.json]: @jmikola The project you mention has probably much less attention on less used versions. But I understand you're not into semantic versioning argument.
